cloudbuild.yaml

```diff
@@ -279,67 +268,77 @@ steps:
+    #!/usr/bin/env bash
 
-#
-# Run the e2e tests without FeatureGates
-#
-
-- name: 'e2e-runner'
-  args:
-    - ''
-    - 'generic'
-    - 'e2e-test-cluster'
-    - 'us-west1-c'
-    - "${_REGISTRY}"
-  id: e2e-stable
-  waitFor:
-    - e2e-feature-gates
-
-#
-# Run the e2e tests on GKE Autopilot with FeatureGates inverted compared to Stable
-#
-# Keep in sync with (the inverse of) pkg/util/runtime/features.go:featureDefaults
-
-- name: 'e2e-runner'
-  args:
-    - 'CustomFasSyncInterval=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&SafeToEvict=true&PodHostname=true&SplitControllerAndExtensions=true&Example=true'
-    - 'gke-autopilot'
-    - 'gke-autopilot-e2e-test-cluster-1-24'
-    - 'us-west1'
-    - "${_REGISTRY}"
-  id: e2e-feature-gates-gke-autopilot
-  waitFor:
-    - e2e-wait-to-become-leader
+    on_going_builds=($(gcloud builds list --ongoing --filter "tags:'e2e-test'" --format="value(id)" | xargs))
+    for on_going_build in ${on_going_builds[@]}
+    do
+      echo "Cancelling e2e test build $on_going_build"
+      gcloud builds cancel $on_going_build
+    done
 
-#
-# Run the e2e tests on GKE Autopilot without FeatureGates, except SafeToEvict and SplitControllerAndExtensions, which we need on Autopilot
-#
+# kick off the child e2e test cloud builds in parallel, fail this build if any of the child build fails
 
-- name: 'e2e-runner'
-  args:
-    - 'SafeToEvict=true&SplitControllerAndExtensions=true'
-    - 'gke-autopilot'
-    - 'gke-autopilot-e2e-test-cluster-1-24'
-    - 'us-west1'
-    - "${_REGISTRY}"
-  id: e2e-stable-gke-autopilot
+- name: 'gcr.io/cloud-builders/gcloud'
+  id: submit-e2e-test-cloud-build
+  entrypoint: bash
+  args: 
+   - '-c'
+   - |
+      set -e
+      pids=()
+      cloudProducts=("generic" "gke-autopilot")
+      for cloudProduct in ${cloudProducts[@]}
+      do
+        if [ $cloudProduct = generic ]
+        then 
+          featureWithGate="CustomFasSyncInterval=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&SafeToEvict=true&PodHostname=true&SplitControllerAndExtensions=true&Example=true"
+          featureWithoutGate=""
+          testClusterLocation="us-west1-c"
+          testCluster="e2e-test-cluster"
+        else
+          featureWithGate="CustomFasSyncInterval=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&SafeToEvict=true&PodHostname=true&SplitControllerAndExtensions=true&Example=true"
+          featureWithoutGate="SafeToEvict=true&SplitControllerAndExtensions=true"
+          testClusterLocation="us-west1"
+          testCluster="gke-autopilot-e2e-test-cluster-1-24"
+        fi
+        # TODO: Add more versions in the list, and generate the test cluster name in the loop rather than hardcode once the clusters are created
```

Comment:
As a callout that I'm sure will be a nuisance here, we can't support Autopilot on <1.24 yet. (I need to add it to the User Guide, woops.)

Reply:
I see, I will keep in mind when revise this part.